### PR TITLE
FIX (CORE/DB/SAI) [Quest] - Weapons of Spirit / Allows Value 3 for react SAI WP Start Consolidated PR

### PR DESCRIPTION
**Pull request by acidmanifesto**
Refactored from tc
This is a refactor from TC to deal with react state 3 invalid failure\error message dealing with SMART_ACTION_WP_START (19). This is in support of PR https://github.com/azerothcore/azerothcore-wotlk/pull/8923

Per Malcrom's advice, Brought over his SQL quest fix change as this pr is what we need with the core changes applied to have it pass compile testing. Consolidating his PR https://github.com/azerothcore/azerothcore-wotlk/pull/8923 with this one.

## Changes Proposed:
-  Refactor how to handle the react value of 3 from tc.
- Add missing script to Witch Doctor Uzer'i in Feralas.
- Consolidates PR https://github.com/azerothcore/azerothcore-wotlk/pull/8923 into this one.
## Closes:
- https://github.com/azerothcore/azerothcore-wotlk/issues/8917

## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/runs/4092916912?check_suite_focus=true

## SOURCE:
https://github.com/TrinityCore/TrinityCore/blob/3e6a4e0b4484b241a103a912ebc180b163fe815c/src/server/game/AI/SmartScripts/SmartScriptMgr.cpp#L1796

## Tests Performed:
Builds and compiles locally with no issues.

## How to Test the Changes:
Need Compile Testing or see if errors are reported on world screen.
then test quest https://classic.wowhead.com/quest=3129/weapons-of-spirit to see if the role play with the Witch Doctor Uzer'i  npc occurs.

## Known Issues and TODO List:

- [x] Test PR https://github.com/azerothcore/azerothcore-wotlk/pull/8923 with this PR
- [x] Is error being reported in world console fix?
- [x] Is compile test failure fix?
- [x] Does @malcrom girth and length exceeds mines?

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by malcrom** [comment]
Yes it does exceed yours.

**Comment by acidmanifesto** [comment]
> Yes it does exceed yours.

thats why the box is checked.
image

**Comment by acidmanifesto** [comment]
Need review

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -1745,8 +1745,8 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                     if (!stored)
                         delete targets;
                 }
-
-                me->SetReactState((ReactStates)e.action.wpStart.reactState);
+                if (e.action.wpStart.reactState <= REACT_AGGRESSIVE)
+                    me->SetReactState((ReactStates)e.action.wpStart.reactState);
```
Comment (Nefertumm) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:1749]: add brackets here

File: data/sql/updates/db_world/rev_1635936745734477829.sql [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+(811503, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 82, 3, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Witch Doctor Uzer''i - On Script - Add Npc Flag Questgiver+Gossip');
+-- Add Waypoints for Witch Doctor Uzer'i
+DELETE FROM `waypoints` WHERE `entry`=8115;
+INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES
+(8115, 1, -4375.1, 259.339, 26.584, 'Witch Doctor Uzer''i'),
+(8115, 2, -4378.28, 255.898, 26.3856, 'Witch Doctor Uzer''i'),
+(8115, 3, -4383, 256.683, 25.8833, 'Witch Doctor Uzer''i'),
+(8115, 4, -4378.28, 255.898, 26.3856, 'Witch Doctor Uzer''i'),
+(8115, 5, -4375.1, 259.339, 26.584, 'Witch Doctor Uzer''i'),
+(8115, 6, -4375.68, 269.717, 25.4947, 'Witch Doctor Uzer''i');
```
Comment (Nefertumm) [inline data/sql/updates/db_world/rev_1635936745734477829.sql:43]: add a new line

**Review by Nefertumm — CHANGES_REQUESTED** [review]
Seems ok, just match the codestyle

**Comment by Nefertumm** [comment]
Just for references, ported from https://github.com/TrinityCore/TrinityCore/commit/60379cf2c15d9cee1414fbaec2b217b89b8dc67e

File: src/server/game/AI/SmartScripts/SmartScript.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/AI/SmartScripts/SmartScript.cpp]: done

File: data/sql/updates/db_world/rev_1635936745734477829.sql (hunk 2 above)
Comment (acidmanifesto) [reply data/sql/updates/db_world/rev_1635936745734477829.sql]: done

**Comment by acidmanifesto** [comment]
## How to Test the Changes:
Need Compile Testing or see if errors are reported on world screen.
then test quest https://classic.wowhead.com/quest=3129/weapons-of-spirit to see if the role play with the Witch Doctor Uzer'i  npc occurs.

**Review by Nefertumm — APPROVED** [review]

**Pull request merged**